api/apps/api/src/modules/scenarios/marxan-run/input-parameter-file.provider.ts

```diff
@@ -78,13 +70,15 @@ class InputParameterFile {
 @Injectable()
 export class InputParameterFileProvider {
   constructor(
-    private readonly scenariosService: ScenariosCrudService,
+    @InjectRepository(Scenario)
+    private readonly scenarioRepository: Repository<Scenario>,
     @Inject(ioSettingsToken)
     private readonly ioSettings: IoSettings,
   ) {}
 
   async getInputParameterFile(scenarioId: string): Promise<string> {
-    const scenario = await this.scenariosService.getById(scenarioId);
+    const scenario = await this.scenarioRepository.findOne(scenarioId);
+    assertDefined(scenario);
```

Comment:
shouldn't (isn't it already?) it be done on service layer?
`api/apps/api/src/modules/scenarios/scenarios.service.ts`

Reply:
it is but I needed to narrow the type here, as `findOne` returns `| undefined`.

On the other hand, it's not in a transaction so it's possible that the assertion will actually throw :)